File: src_front_end/src/services/ShainService.js
import http from '@/http-common';

const ShainService = {

  getInformation() {
    return http.get('/shain/information');
  },
  listShiansForApproval(stepNumber, bumonCd, requestCd) {
    return http.get(`/shain/get-shain-for-approval?stepNumber=${stepNumber}&bumonCd=${bumonCd}&requestCd=${requestCd}`)
  },
  updateShainInformation(shainResponse) {
    return http.post("/shain/update", shainResponse);
  }
}

export default ShainService;

File: src/main/java/sskf/service/ShainService.java
package sskf.service;

import org.springframework.data.domain.Page;
import sskf.model.basemodel.BaseSearchRequest;
import sskf.model.entity.ShainEntity;
import sskf.model.response.ShainResponse;

import java.util.List;

public interface ShainService {

    ShainResponse getInformation();

    Page<ShainResponse> listShains(BaseSearchRequest baseSearchRequest);

    Page<ShainResponse> listShainsMaster(BaseSearchRequest baseSearchRequest);

    String getUserNameByToken();

    String getBumonByToken();

    List<ShainEntity> getListShains(BaseSearchRequest baseSearchRequest);

    ShainEntity getLoggedInShainEntity();

    List<ShainResponse> listShainForApproval(Byte stepNumber, String bumonCd, Long requestCd);

    String getCurrentUserBumonCd ();

    ShainResponse updateShainInformation(ShainResponse shainResponse);

}
